controllers/azureidentity_controller.go

```diff
@@ -43,24 +35,36 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+	infraexpv1 "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1beta1"
+	"sigs.k8s.io/cluster-api-provider-azure/feature"
+	"sigs.k8s.io/cluster-api-provider-azure/util/reconciler"
+	"sigs.k8s.io/cluster-api-provider-azure/util/system"
+	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
 )
 
 // AzureIdentityReconciler reconciles Azure identity objects.
 type AzureIdentityReconciler struct {
 	client.Client
-	Log              logr.Logger
 	Recorder         record.EventRecorder
 	ReconcileTimeout time.Duration
 	WatchFilterValue string
 }
 
 // SetupWithManager initializes this controller with a manager.
 func (r *AzureIdentityReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
-	log := r.Log.WithValues("controller", "AzureIdentity")
+	_, log, done := tele.StartSpanWithLogger(ctx,
+		"controllers.AzureIdentityReconciler.SetupWithManager",
+		tele.KVP("controller", "AzureIdentity"),
```

Comment:
same comment as above for passing key value pairs, and for other occurrences below.